internal/server/data/identity.go

```diff
@@ -155,6 +155,21 @@ func AssignIdentityToGroups(tx WriteTxn, user *models.ProviderUser, newGroups []
 		}
 	}
 
+	if count := len(idsToRemove) + len(addIDs); count > 0 {
```

Comment:
It feels like the logic above should have used a function like `AddUsersToGroup` rather than manually needing to check if we need to update the index. That could be a future refactor though.

Reply:
Ya, I was considering that. I think I didn't take that approach because I thought there might be some subtle difference in here. I think one difference is that `AddUserstoGroup` is single group many users, where as this is single user many groups.  We can likely share more code here, but I didn't want to do that as part of this change.